src/oss-detect-cryptography/DetectCryptographyTool.cs

```diff
@@ -232,23 +229,14 @@ public DetectCryptographyTool()
         /// </summary>
         /// <param name="purl">The package-url of the package to analyze.</param>
         /// <returns>List of tags identified</returns>
-        public async Task<List<IssueRecord>> AnalyzePackage(PackageURL purl)
+        public async Task<List<IssueRecord>> AnalyzePackage(PackageURL purl, string targetDirectoryName)
         {
             Logger.Trace("AnalyzePackage({0})", purl.ToString());
 
             var analysisResults = new List<IssueRecord>();
 
-            string targetDirectoryName = null;
-            while (targetDirectoryName == null || Directory.Exists(targetDirectoryName))
-            {
-                targetDirectoryName = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
-            }
-
-            Logger.Trace("Creating directory [{0}]", targetDirectoryName);
-            Directory.CreateDirectory(targetDirectoryName);
-
-            var downloadTool = new DownloadTool();
-            var directoryNames = await downloadTool.Download(purl, targetDirectoryName);
+            var packageDownloader = new PackageDownloader(purl, (string)(this.Options["cache-directory"]));
```

Comment:
Bug? You don't use the `targetDirectoryName` argument and instead pull `cache-directory` out of options.